### DISCUSSION TITLE
What is the difference between useSendTransaction and useWriteContract?

### DISCUSSION THREAD

**Discussion by UTkzhang** [wevm/wagmi#3592] [General]
Just wondering why there are these two hook options for sending a wallet client transaction, and whether or not there’s anything in the implementation that would make one of these hooks to be better in certain use cases over the other.

**Comment by thoerner** [comment]
The difference boils down to their specific use cases in interacting with the blockchain:

- `useSendTransaction` is meant for sending ETH from one address to another without the need for interacting with a smart contract. It's straightforward and used primarily for transfers of value.

- `useWriteContract` is used for interacting with smart contracts by calling functions defined within them. You need to provide the contract's ABI, the function name, and any necessary arguments. It's essential for operations that require executing contract functions, such as token minting or participating in a DAO.

In essence, use `useSendTransaction` for simple ETH transfers and `useWriteContract` when you need to execute functions on a smart contract.

**Reply by UTkzhang** [reply]
Thanks for the explanation. This makes sense, but just to further clarify, useContractWrite itself is just a wrapper around a somewhat complex transaction, so you could just send data along with a transaction to a smart contract address, but the hook makes it easier to interact with the smart contracts?
